Orange/widgets/data/owconcatenate.py

```diff
@@ -126,10 +131,12 @@ def __init__(self):
         box.layout().insertSpacing(1, 20)
 
     @check_sql_input
+    @Inputs.primary_data
```

Comment:
Gosh, this is so annoying. The order in the PR, while essentially incorrect, works because `@Inputs.<name>` decorator does not really wrap the function but just stores its name. On the other hand, the (essentially) correct order does not work because the `check_sql_input` decorator changes the method's `__name__` to `new_f`, and so `@Inputs` stores `new_f` as the signal handler.

Please change the order anyway, so we're prepared for potential future changes in signal handling. To prevent crashes, fix the `check_sql_input` decorator. In the definition of the decorator, decorate `new_f` with `wraps`, like this:

    @wraps(f)
    def new_f(widget, data, *args, **kwargs):

This will rename `new_f` to the name of the wrapped method, so `@Inputs` will see the correct name.